src/Build/Globbing/MSBuildGlob.cs

```diff
@@ -171,7 +171,7 @@ public static MSBuildGlob Parse(string globRoot, string fileSpec)
             ErrorUtilities.VerifyThrowArgumentNull(fileSpec, nameof(fileSpec));
             ErrorUtilities.VerifyThrowArgumentInvalidPath(globRoot, nameof(globRoot));
 
-            if (globRoot == string.Empty)
+            if (string.IsNullOrEmpty(globRoot))
```

Comment:
So globRoot is guaranteed to not be null as `ErrorUtilities.VerifyThrowArgumentNull` is called just before this. The change isn't necessary and adds an extra stack frame to what can stay as a simple string check. If anything this should (and all other uses of `X == string.Empty`) be replaced with `globRoot.Length == 0`

Reply:
Had to verify for myself. `IsNullOrEmpty` checks `str == null || str.Length == 0`. Since we know, some of these couldn't possibly be null, just checking length would be better.